File: SimpleHabit/app/src/main/java/com/me/simplehabit/adapters/MainViewPagerAdapter.java
package com.me.simplehabit.adapters;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.me.simplehabit.fragments.OnTheGoFragment;
import com.me.simplehabit.fragments.SeriesFragment;
import com.me.simplehabit.fragments.TeachersFragment;

public class MainViewPagerAdapter extends FragmentPagerAdapter {

    public MainViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        if (position==0){
            return OnTheGoFragment.getInstance();
        }else if (position==1){
            return SeriesFragment.getInstance();
        }else {
            return TeachersFragment.getInstance();
        }
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        if (position==0){
            return "ON THE GO";
        }else if (position==1){
            return "SERIES";
        }else {
            return "TEACHERS";
        }
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/adapters/MainViewPagerAdapter.kt
package com.me.simplehabit.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.me.simplehabit.fragments.OnTheGoFragment
import com.me.simplehabit.fragments.SeriesFragment
import com.me.simplehabit.fragments.TeachersFragment

class MainViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> OnTheGoFragment.newInstance()
            1 -> SeriesFragment.newInstance()
            else -> TeachersFragment.newInstance()
        }
    }

    override fun getCount(): Int {
        return 3
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> "ON THE GO"
            1 -> "SERIES"
            else -> "TEACHERS"
        }
    }
}
